x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -188,6 +221,12 @@ export const TimelineDataTableComponent: React.FC<DataTableProps> = memo(
       [tableRows, handleOnEventDetailPanelOpened, handleOnPanelClosed]
     );
 
+    useEffect(() => {
+      if (isTimelineExpandableFlyoutEnabled && !isTimelineExpandableFlyoutOpen) {
```

Comment:
need to test it, but not sure if it would also be smart to cache the previous open state in a ref to compare  so something else that fires in this file doesn't trigger the  `handleOnPanelClosed`. Probably doesn't matter though as the `close fn` doesn't do anything crazy 🤷🏾‍♂️

Reply:
@michaelolo24 , you are right. I did some changes now which transfers the responsibility to run `onCloseFlyout` effect on the `expandable flyout` hook. It makes sure that `onClose` runs only and only once. See if this change makes sense.